lib/IRGen/IRGenSIL.cpp

```diff
@@ -1163,8 +1163,42 @@ class IRGenSILFunction :
           emitPackCountDebugVariable(Shape);
         }
       });
+
+    if (auto *BGT = llvm::dyn_cast<BoundGenericType>(Ty)) {
+      auto Decl = BGT->getDecl();
+      auto GE = Decl->getGenericEnvironment();
+      auto Requirements = BGT->getDecl()
+                              ->getGenericEnvironment()
+                              ->getGenericSignature()
+                              .getRequirements();
+      for (auto Requirement : Requirements) {
+        if (Requirement.getKind() == RequirementKind::Conformance) {
+          auto ProtocolDecl = Requirement.getProtocolDecl();
+          auto ConformingType = Requirement.getFirstType();
+          Type Archetype;
+          if (auto GTPT = llvm::dyn_cast<GenericTypeParamType>(
```

Comment:
These conditionals are unnecessary. mapTypeIntoContext() takes a Type. Also, we use `ConformingType->getAs<...>` and not `llvm::dyn_cast<>(ConformingType.getPointer())`